src/v2/Apps/Auction/Components/Form/helpers.tsx

```diff
@@ -50,3 +52,39 @@ export const determineDisplayRequirements = (
     requiresPaymentInformation: !(isRegistered || me.hasQualifiedCreditCards),
   }
 }
+
+export function createStripeWrapper<T>(Component: React.FC<T>): React.FC<T> {
+  const StripeInjectedComponent = injectStripe(Component)
+
+  return props => {
+    const [stripe, setStripe] = useState(null)
+
+    function setupStripe() {
+      setStripe(window.Stripe(sd.STRIPE_PUBLISHABLE_KEY))
```

Comment:
I don't completely understand these two **[rules of hooks](https://reactjs.org/docs/hooks-rules.html#only-call-hooks-at-the-top-level)** but wonder if they apply here.
- Only Call Hooks at the Top Level
- Only Call Hooks from React Functions

Reply:
Hmm it is definitely not conventional to use a state this way. Although I'm not sure how we could make it better.